src/utils/string.formatter.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * String formatting functions
+ */
+const REGEX_NON_PRINTABLE_CHARS: RegExp = /[^ -\xFF€]+/g;
```

Comment:
would the regex `/[^\x00-\x7F]/g` be better? (match all non ASCII characters)
This regex seems to be match one or more, SPACE to ÿ, and € why this range specifically?

Reply:
I'm not sure we want to exclude all non ascii characters and I think we want to start at 32 DEC rather than 0 so that we don't have \n etc.

Reply:
is there any guidance on what is a "Non printable character"?

Reply:
Good question

Reply:
I've found this? https://en.wikipedia.org/wiki/Non-printing_character_in_word_processors

Reply:
Looked here https://web.itu.edu.tr/sgunduz/courses/mikroisl/ascii.html and foudn that non printable is 0-32 and printable is the extensed ascii set 32-255.

Reply:
Looking at that then, and the wikipedia article, should the regex match 0-31 (inclusive), so when calling str.replace it removes those characters?

Reply:
From what I remember from the extension work I think we want to included extended ascii - i'll see if I can find the original ticket for that.

Reply:
May be worth flagging the story and asking for more detailed requirements? It shouldn't really be down to us to decide which characters are or aren't "printable"

Reply:
To address your question are we suggesting an explict positive reference to 0-31 rather than anything that is not 32-255?

Reply:
If the definition of a non printable character is 0-31. Then yes we should only match and remove 0-31.
As Not 32-255 would remove characters over 255, which if definition for non printable is 0-31, then over 255 is still printable.
We need a more concrete definition of what the story considers to be a non printable character

Reply:
I think the current definition is specifically to limit this to just the ascii and extended ascii set - but it would be worth verifying this.

Reply:
Just a final note - I'm guessing the reason we used this for extensions is so that we don't leave ourselves too exposed to any potential complications that a  >255 might bring.